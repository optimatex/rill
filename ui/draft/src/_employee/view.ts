export {
  React,
  useState,
  useCallback,
  css,
  styled,
  createGlobalStyle,
} from '@rill/view';
