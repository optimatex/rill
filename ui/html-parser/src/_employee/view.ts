export { React, RC } from '@rill/view';
