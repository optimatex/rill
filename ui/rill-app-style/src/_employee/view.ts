export { createGlobalStyle, css } from '@rill/view';
