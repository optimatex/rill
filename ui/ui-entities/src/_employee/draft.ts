export { TextEditor } from '@rill/draft';
