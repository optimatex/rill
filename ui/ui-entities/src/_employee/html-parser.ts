export { htmlParser } from '@rill/html-parser';
