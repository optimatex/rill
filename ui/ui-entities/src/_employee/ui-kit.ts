export {
  // Layout
  Row,
  ListContainer,
  // Card
  Card,
  Post,
  Panel,
  // Menu
  MenuItemLink,
  // Typography
  Paragraph,
  Name,
  // Image
  Image,
  ImageCropper,
  ImagesLayout,
  // Video
  YoutubeVideos,
  // Content
  RichContentContainer,
} from '@rill/ui-kit';
