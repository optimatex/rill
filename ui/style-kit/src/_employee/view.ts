import { css as originalCss } from '@rill/view';

export const css = originalCss;
