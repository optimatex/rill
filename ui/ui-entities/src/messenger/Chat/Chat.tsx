import { React } from '_employee/view';
import { Post } from '_employee/ui-kit';

import { ICommonMessageProps } from '@types';
import Messages from '../Messages';
import MessageForm from '../MessageForm';

const Chat = (props: ICommonMessageProps) => (
  <Post title="Chat room">
    <Messages messages={props.messages} />

    <MessageForm onAddMessage={props.onAddMessage} />
  </Post>
);

export default Chat;
