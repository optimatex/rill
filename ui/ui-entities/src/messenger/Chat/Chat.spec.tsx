import { React } from '_employee/view';
import { render, mockMessages_1 } from 'test';
import Chat from '.';

describe('Chat', () => {
  it('should render without errors', () => {
    const { container } = render(
      <Chat messages={mockMessages_1} onAddMessage={jest.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });
});
