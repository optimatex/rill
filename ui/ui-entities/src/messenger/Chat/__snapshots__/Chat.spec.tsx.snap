// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat should render without errors 1`] = `
.c6 {
  object-fit: cover;
  max-width: 100%;
  display: block;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  height: 30px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto;
  width: 30px;
  overflow: hidden;
  border-radius: 50px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  box-sizing: border-box;
  padding: 5px 8px;
}

.c4 > * {
  margin-right: 16px;
}

.c4 > *:last-child {
  margin-right: 0;
}

.c4 > * {
  margin-right: 12px;
}

.c2 {
  margin: 0;
  line-height: 1rem;
  font-family: 'open-sans-regular';
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: border-box;
  padding: 4px 8px;
  border-width: 1px;
  border-style: solid;
  background: red;
  color: white;
  padding-left: 8px;
  padding-right: 8px;
}

.c8 {
  font-size: 8px;
  font-family: roboto;
  font-weight: bolder;
}

.c7 p {
  padding: 0;
  margin: 0;
}

.c9 {
  border-width: 1px;
  border-style: solid;
  border-width: 0;
  border-top-width: 1px;
}

.c0 {
  padding: 0;
  border-width: 1px;
  border-style: solid;
  background: red;
  color: white;
}

.c0 > *:nth-child(1) {
  border-radius: 4px 4px 0 0;
}

.c0 > *:last-child {
  border-radius: 0 0 4px 4px;
}

.c1 {
  box-sizing: border-box;
  padding: 5px 8px;
  border-bottom-width: 1px;
  border-bottom-style: solid;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  list-style-type: none;
  padding: 0;
  margin: 0;
}

.c3 > * {
  margin-bottom: px;
}

.c3 > *:last-child {
  margin-bottom: 0;
}

.c10 {
  width: 100%;
  padding: 10px;
}

.c10 .public-DraftEditor-content {
  margin-bottom: 10px;
}

<div>
  <div
    class="c0"
    data-testid="post"
  >
    <div
      class="c1"
    >
      <h3
        class="c2"
      >
        Chat room
      </h3>
    </div>
    <ul
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <img
            class="c6"
            src="https://i.pravatar.cc/60"
          />
        </div>
        <div
          class="c7"
        >
          <p
            class="c8"
          >
            I've been a QA engineer with comp.sci background working in many places building quality tools for over 11 years. Last year during the pandemic most of my company was fired, including myself. So, I decided to apply for a web developer role at Napster. This video was literally all I used to do the technical exam. I am now finishing my probation period! Thanks, mate!
          </p>
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <img
            class="c6"
            src="https://i.pravatar.cc/62"
          />
        </div>
        <div
          class="c7"
        >
          <p
            class="c8"
          >
            Who else was feeling so hopeful that he would see that he used ‚Äúd‚Äù instead of ‚Äúid‚Äù on one of the json server entries. 
Btw I‚Äôm completely new to React and this tutorial has been really great! Really easy to follow, well explained, and just focusing on fundamentals; exactly what I expect in a crash course üëç Keep it up!
          </p>
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <img
            class="c6"
            src="https://i.pravatar.cc/63"
          />
        </div>
        <div
          class="c7"
        >
          <p
            class="c8"
          >
            Please make a tutorial about Spring. Thank you so much, you are a legend. :D
          </p>
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <img
            class="c6"
            src="https://i.pravatar.cc/64"
          />
        </div>
        <div
          class="c7"
        >
          <p
            class="c8"
          >
            One small thing I want to mention is around 33:20 when we add the icons, you will see a div with the class of ‚Äúcard-body‚Äù just pop in there. That got cut out for some reason, so just be sure to add that. Thanks for watching guys
          </p>
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <img
            class="c6"
            src="https://i.pravatar.cc/65"
          />
        </div>
        <div
          class="c7"
        >
          <p
            class="c8"
          >
            glad to hear your voice and see how great teacher you are. Btw I‚Äôm completely new to React and this tutorial has been really great!
          </p>
        </div>
      </div>
    </ul>
    <div
      class="c9"
    >
      <div
        class="c10"
        data-testid="text-editor-wrap"
      >
        <div
          class="DraftEditor-root"
        >
          <div
            class="public-DraftEditorPlaceholder-root"
          >
            <div
              class="public-DraftEditorPlaceholder-inner"
              id="placeholder-1"
              style="white-space: pre-wrap;"
            >
              Type your message here...
            </div>
          </div>
          <div
            class="DraftEditor-editorContainer"
          >
            <div
              aria-describedby="placeholder-1"
              class="notranslate public-DraftEditor-content"
              contenteditable="true"
              role="textbox"
              spellcheck="false"
              style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
            >
              <div
                data-contents="true"
              >
                <div
                  class=""
                  data-block="true"
                  data-editor="1"
                  data-offset-key="1-0-0"
                >
                  <div
                    class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                    data-offset-key="1-0-0"
                  >
                    <span
                      data-offset-key="1-0-0"
                    >
                      <br
                        data-text="true"
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="c11"
        >
          Send
        </button>
      </div>
    </div>
  </div>
</div>
`;
